### PR TITLE
add df-other-vectors-type

### PR DESCRIPTION
**Pull request by BenLubar**
see DFHack/DFHack#1558

### CODE REVIEW

File: df.building-vectors.xml [hunk 1]
```diff
@@ -379,6 +379,10 @@
             <item-attr name='building' value='Instrument'/>
         </enum-item>
     </enum-type>
+
+    <df-other-vectors-type type-name='buildings_other' index-enum='buildings_other_id' item-type='building'>
+        <stl-vector name='BED' pointer-type='building_bedst'/>
```
Comment (cvuchener) [inline df.building-vectors.xml:384]: What does it mean when df-other-vectors-type has a child? Is it added after the other vectors?

Comment (BenLubar) [reply df.building-vectors.xml:384]: It replaces the vector by the same name. Only stl-vector is allowed as a child of df-other-vectors-type.

Comment (lethosor) [reply df.building-vectors.xml:410]: Ok, so it's overriding what the BED vector would otherwise have been? (Why does just this one need to be overridden?)

**Comment by cvuchener** [comment]
Just an idea: if you end up overriding most vector fields, wouldn't it be better to do it the other way? I mean explicitly defining all df-other-vectors-type fields, then generating then enum from the fields. There would be less duplicated information.

File: df.building-vectors.xml (hunk 1 above)
Comment (BenLubar) [reply df.building-vectors.xml:410]: that was my test. the PR now includes all of the vectors that have defined subclasses.

**Pull request merged**